### PR TITLE
Add endpoint for posting feedback

### PR DESCRIPTION
**Pull request by hankewyczz**
# Purpose

The feedback/contact buttons do not work. This adds a feedback endpoint to the notifs server, which notifies the Search team on Slack.

<br>

# Tickets

https://trello.com/c/MzKg5kJT/288-feedback-buttons-are-inactionable

# Contributors

- @hankewyczz 

<br>

# Checklist

- [x] Filled out PR template :wink:
- [ ] Approved by designers
- [ ] Is passing linting checks
- [ ] Is passing tsc
- [ ] Is passing existing tests
- [ ] Has documentation and comments in code
- [ ] Has test coverage for code
- [ ] Will have clear squash commit message

<br>
@sandboxnu/searchneu

### CODE REVIEW

**Comment by hankewyczz** [comment]
When merging, the environment variable `SLACK_WEBHOOK_URL` must be added to the prod task definition in AWS. The variable is already in the Parameter Store.

**Review by andrewydai — CHANGES_REQUESTED** [review]
I think this looks good codewise, but wondering if there is a need to make a request to CCA, and then to slack. Is there an issue with keeping the webhook on the front end env variables and issuing the request to slack from there? It would also stop us from adding an endpoint to our twilio server that doesn't seem relevant to notifs (I may be missing a security issue here?)

**Comment by hankewyczz** [comment]
@andrewydai 

I'm split on this. My main reasoning for this was security, but it may be premature.

The Slack webhook has no authentication - the URL itself is the secret

> Keep it secret, keep it safe. Your webhook URL contains a secret. Don't share it online, including via public version control repositories. Slack actively searches out and revokes leaked secrets.
- https://api.slack.com/messaging/webhooks#getting_started


For that reason, I wasn't a fan of sending the request thru the frontend, as anyone who opens their console/network logs can see it. On the other hand, all they can do with the URL is spam the support channel - which they can do anyways. In either case, we can always revoke the webhook and create a new one, which is trivial. 

Basically, what it came down to for me was - if someone starts abusing it, we can add some rate-limiting controls on the backend. If someone is persistently spamming the channel (ie. using the new links as soon as we generate them), this would be a way of protecting against that. To send the request from the frontend, we'd have to expose the URL, which isn't great (I'm not really worried about people abusing this, but more likely a scraper finding it). 

I agree with you on the server issue - I don't like using the notifs server for this, but at the same time, spinning up another server for this would be overkill.

**Comment by megandouli** [comment]
> @andrewydai
> 
> I'm split on this. My main reasoning for this was security, but it may be premature.
> 
> The Slack webhook has no authentication - the URL itself is the secret
> 
> > Keep it secret, keep it safe. Your webhook URL contains a secret. Don't share it online, including via public version control repositories. Slack actively searches out and revokes leaked secrets.
> 
> * https://api.slack.com/messaging/webhooks#getting_started
> 
> For that reason, I wasn't a fan of sending the request thru the frontend, as anyone who opens their console/network logs can see it. On the other hand, all they can do with the URL is spam the support channel - which they can do anyways. In either case, we can always revoke the webhook and create a new one, which is trivial.
> 
> Basically, what it came down to for me was - if someone starts abusing it, we can add some rate-limiting controls on the backend. If someone is persistently spamming the channel (ie. using the new links as soon as we generate them), this would be a way of protecting against that. To send the request from the frontend, we'd have to expose the URL, which isn't great (I'm not really worried about people abusing this, but more likely a scraper finding it).
> 
> I agree with you on the server issue - I don't like using the notifs server for this, but at the same time, spinning up another server for this would be overkill.

@hankewyczz @andrewydai FWIW last winter SearchNEU had a marathon to split the GraphQL server (aka course-catalog-api) from the notifications server. Unfortunately, when we switched to SMS notifs, we reverted the split and once again coupled CCA with the SearchNEU backend.

**Review by megandouli — APPROVED** [review]
just want to note that deploying this change will be non-trivial because we have a new env variable so we'll need to use Terraform @andrewydai happy to pair on this sometime if you'd like

**Review by andrewydai — APPROVED** [review]
